pkg/authentication/externalidp/awsiam/aws_client.go

```diff
@@ -0,0 +1,194 @@
+	"github.com/aws/smithy-go/middleware"
+	smithyhttp "github.com/aws/smithy-go/transport/http"
+	"github.com/google/uuid"
+)
+
+var ErrInvalidCredentialsFormat = errors.New("missing required parts in query param X-Amz-Credential")
+
+const (
+	AuthVersion       = "2011-06-15"
+	AuthMethod        = http.MethodPost
+	AuthAction        = "GetCallerIdentity"
+	AuthAlgorithm     = "AWS4-HMAC-SHA256"
+	StsGlobalEndpoint = "sts.amazonaws.com"
+	AuthActionKey     = "Action"
+	AuthVersionKey    = "Version"
+	AuthAlgorithmKey  = "X-Amz-Algorithm"
+	//nolint:gosec
+	AuthCredentialKey = "X-Amz-Credential"
+	AuthDateKey       = "X-Amz-Date"
+	AuthExpiresKey    = "X-Amz-Expires"
+	//nolint:gosec
+	AuthSecurityTokenKey = "X-Amz-Security-Token"
+	AuthSignedHeadersKey = "X-Amz-SignedHeaders"
+	AuthSignatureKey     = "X-Amz-Signature"
+	DatetimeFormat       = "20060102T150405Z"
+	CredentialTimeFormat = "20060102"
+)
+const (
+	DefaultSTSLoginExpire  = 15 * time.Minute
+	DefaultRefreshInterval = 5 * time.Minute
+	DefaultURLPresignTTL   = 1 * time.Minute
+	DefaultTokenTTL        = 3600 * time.Minute
+	minLengthSplitCreds    = 3
+)
+
+type AWSIdentityTokenInfo struct {
+	Method             string   `json:"method"`
+	Host               string   `json:"host"`
+	Region             string   `json:"region"`
+	Action             string   `json:"action"`
+	Date               string   `json:"date"`
+	ExpirationDuration string   `json:"expiration_duration"`
+	AccessKeyID        string   `json:"access_key_id"`
+	Signature          string   `json:"signature"`
+	SignedHeaders      []string `json:"signed_headers"`
+	Version            string   `json:"version"`
+	Algorithm          string   `json:"algorithm"`
+	SecurityToken      string   `json:"security_token"`
+}
+
+type IAMAuthParams struct {
+	ProviderType        string
+	TokenRequestHeaders map[string]string
+	URLPresignTTL       time.Duration
+	TokenTTL            time.Duration
+	RefreshInterval     time.Duration
+}
+
+type IAMAuthParamsOptions = func(params *IAMAuthParams)
+
+func WithProviderType(provider string) IAMAuthParamsOptions {
+	return func(params *IAMAuthParams) {
+		params.ProviderType = provider
+	}
+}
+func WithRefreshInterval(refreshInterval time.Duration) IAMAuthParamsOptions {
+	return func(params *IAMAuthParams) {
+		params.RefreshInterval = refreshInterval
+	}
+}
+func WithURLPresignTTL(urlPresignTTL time.Duration) IAMAuthParamsOptions {
+	return func(params *IAMAuthParams) {
+		params.URLPresignTTL = urlPresignTTL
+	}
+}
+func WithTokenTTL(tokenTTL time.Duration) IAMAuthParamsOptions {
+	return func(params *IAMAuthParams) {
+		params.TokenTTL = tokenTTL
+	}
+}
+func WithTokenRequestHeaders(tokenRequestHeaders map[string]string) IAMAuthParamsOptions {
+	return func(params *IAMAuthParams) {
+		params.TokenRequestHeaders = tokenRequestHeaders
+	}
+}
+
+func NewIAMAuthParams(lakeFSHost string, opts ...IAMAuthParamsOptions) *IAMAuthParams {
+	headers := map[string]string{
+		"X-LakeFS-Server-ID": lakeFSHost,
```

Comment:
please export to global-const so that it can be re-used by everest and other tools as well as verified in tests. 
i.e something like: 

```go
const (
   // Header that should contain the lakeFS host and will signed as part of the identity token request by default
    HostServerIDHeader ...
)
```